pkg/proto/v0/accounts.pb.micro_test.go

```diff
@@ -0,0 +1,944 @@
+		"  ",
+	}
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	for _, id := range invalidIds {
+		req := &proto.DeleteGroupRequest{Id: id}
+		res, err := cl.DeleteGroup(context.Background(), req)
+		assert.IsType(t, &empty.Empty{}, res)
+		assert.Empty(t, res)
+		assert.Error(t, err)
+		assert.Equal(
+			t,
+			fmt.Sprintf("{\"id\":\".\",\"code\":404,\"detail\":\"could not read group: open accounts-store/groups/%v: no such file or directory\",\"status\":\"Not Found\"}", id),
+			err.Error(),
+		)
+	}
+	cleanUp(t)
+}
+
+func TestDeleteGroupInvalidId(t *testing.T) {
+	invalidIds := map[string]string{
+		".":                                     ".",
+		"hello/world":                           "hello/world",
+		"/new-id":                               "/new-id",
+		"/new-id/":                              "/new-id",
+		"/0ed84f08-aa0a-46e4-8e42-f0a5d6e1b059": "/0ed84f08-aa0a-46e4-8e42-f0a5d6e1b059",
+		"":                                      ".",
+	}
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	for id, val := range invalidIds {
+		req := &proto.DeleteGroupRequest{Id: id}
+		res, err := cl.DeleteGroup(context.Background(), req)
+		assert.IsType(t, &empty.Empty{}, res)
+		assert.Empty(t, res)
+		assert.Error(t, err)
+		assert.Equal(
+			t,
+			fmt.Sprintf("{\"id\":\".\",\"code\":500,\"detail\":\"could not clean up group id: invalid id %v\",\"status\":\"Internal Server Error\"}", val),
+			err.Error(),
+		)
+	}
+	cleanUp(t)
+}
+
+func TestUpdateGroup(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+	createGroup(t, grp1)
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	updateGrp := &proto.Group{
+		Id: grp1.Id,
+	}
+
+	req := &proto.UpdateGroupRequest{Group: updateGrp}
+
+	res, err := cl.UpdateGroup(context.Background(), req)
+
+	assert.IsType(t, &proto.Group{}, res)
+	assert.Empty(t, res)
+	assert.Error(t, err)
+	assert.Equal(
+		t,
+		"{\"id\":\".\",\"code\":500,\"detail\":\"not implemented\",\"status\":\"Internal Server Error\"}",
+		err.Error(),
+	)
+	cleanUp(t)
+}
+
+// https://github.com/owncloud/ocis-accounts/issues/61
+func TestAddMember(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+	account := getAccount("user1")
+
+	updatedGroup := grp1
+	updatedGroup.Members = append(updatedGroup.Members, &proto.Account{Id: account.Id})
+
+	createGroup(t, grp1)
+	createAccount(t, account.PreferredName)
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.AddMemberRequest{GroupId: grp1.Id, AccountId: account.Id}
+
+	res, err := cl.AddMember(context.Background(), req)
+	checkError(t, err)
+
+	assert.IsType(t, &proto.Group{}, res)
+
+	// Should return the group but returns empty
+	// assertGroupsSame(t, updatedGroup, res)
+
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, updatedGroup)
+
```

Comment:
need to check if member was added

Reply:
the updatedGroups contains the new member too. if the response contains the updated group then I think we can safely assume that them member was added.